beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/publisher/BlockPublisherFuluTest.java

```diff
@@ -0,0 +1,203 @@
+import tech.pegasys.teku.validator.coordinator.DutyMetrics;
+
+class BlockPublisherFuluTest {
+  private final DataColumnSidecarGossipChannel dataColumnSidecarGossipChannel =
+      mock(DataColumnSidecarGossipChannel.class);
+  private final CustodyGroupCountManager custodyGroupCountManager =
+      mock(CustodyGroupCountManager.class);
+  private final OptionalInt dasPublishWithholdColumnsEverySlots = OptionalInt.empty();
+  private final BlockPublisherFulu blockPublisherFulu =
+      new BlockPublisherFulu(
+          mock(AsyncRunner.class),
+          mock(BlockFactory.class),
+          mock(BlockImportChannel.class),
+          mock(BlockGossipChannel.class),
+          dataColumnSidecarGossipChannel,
+          mock(DutyMetrics.class),
+          custodyGroupCountManager,
+          dasPublishWithholdColumnsEverySlots,
+          true);
+
+  private final Spec spec = TestSpecFactory.createMinimalFulu();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final List<DataColumnSidecar> dataColumnSidecars =
+      dataStructureUtil.randomDataColumnSidecars();
+
+  @Test
+  void publishBlobSidecars_shouldThrow() {
+    final BlobSidecar blobSidecar = mock(BlobSidecar.class);
+    final List<BlobSidecar> blobSidecars = List.of(blobSidecar);
+    assertThatThrownBy(
+            () ->
+                blockPublisherFulu.publishBlobSidecars(
+                    blobSidecars, BlockPublishingPerformance.NOOP))
+        .isInstanceOf(RuntimeException.class)
+        .hasMessage("Unexpected call to publishBlobSidecars in Fulu");
+  }
+
+  @Test
+  void publishDataColumnSidecars() {
+    blockPublisherFulu.publishDataColumnSidecars(
+        dataColumnSidecars, BlockPublishingPerformance.NOOP);
+
+    verify(dataColumnSidecarGossipChannel)
+        .publishDataColumnSidecars(dataColumnSidecars, RemoteOrigin.LOCAL_PROPOSAL);
+  }
+
+  @Test
+  void mustPublishAll_isAlwaysTrueWhenWithholdIsEmpty() {
+    for (UInt64 i = UInt64.ZERO; i.isLessThan(1_000_000); i = i.increment()) {
+      assertThat(blockPublisherFulu.mustPublishAll(i)).isTrue();
+    }
+  }
+
+  @Test
+  void mustPublishAll_isAlwaysFalseWhenWithholdIsSetBeforeFirstUse() {
+    final BlockPublisherFulu blockPublisherFulu =
+        new BlockPublisherFulu(
+            mock(AsyncRunner.class),
+            mock(BlockFactory.class),
+            mock(BlockImportChannel.class),
+            mock(BlockGossipChannel.class),
+            dataColumnSidecarGossipChannel,
+            mock(DutyMetrics.class),
+            custodyGroupCountManager,
+            OptionalInt.of(10),
+            true);
+    for (UInt64 i = UInt64.ZERO; i.isLessThan(1_000); i = i.increment()) {
+      assertThat(blockPublisherFulu.mustPublishAll(i)).isFalse();
+    }
+  }
+
+  @Test
+  void mustPublishAll_isResetAfterPublish() {
+    final int newWithholdSlots = 10;
+    final BlockPublisherFulu blockPublisherFulu =
```

Comment:
does look like this might be duplicated a bit, we could maybe clean these tests up.